### PR TITLE
(WIP)[Enhancement](Broker load) Show accurate import progress of broker load 

### LINKED ISSUE #17220: [Enhancement] Show accurate import progress of broker load (load csv files)
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

the import progress is not accurate when importing compressed format data (BE). Currently, broker load does not use real size to calculate progress, so when the import progress reaches 99%, it still has more than 1% file has not been uncompressed, this makes load time can not be estimated intuitively. 

### Solution

based on the import file format, calculate accurate import progress

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by 1330571**
# Proposed changes

Issue Number: close #17220

## Problem summary

1. Add a readBytes field in `FrontendService.thrift`
2. If the remote file reader support `read_bytes` feature, use `read_bytes` to replace `load_bytes` when calculating load progress(FE) and report TReportExecStatusParams(BE)
3. Add readbytes logic in `loadstatistics` and BE

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by 1330571** [comment]
run buildall

**Comment by 1330571** [comment]
run buildall

File: be/src/vec/exec/scan/vfile_scanner.cpp [hunk 1]
```diff
@@ -580,6 +580,8 @@ Status VFileScanner::_get_next_reader() {
             _cur_reader.reset(new CsvReader(_state, _profile, &_counter, _params, range,
                                             _file_slot_descs, _io_ctx.get()));
             init_status = ((CsvReader*)(_cur_reader.get()))->init_reader(_is_load);
+            _state->set_support_read_bytes(
```
Comment (yiguolei) [inline be/src/vec/exec/scan/vfile_scanner.cpp:583]: maybe we should add scan_bytes feature to all scan node. For example, broker load, insert into select, stream load and etc.

File: be/src/exec/line_reader.h [hunk 2]
```diff
@@ -28,7 +28,8 @@ class LineReader {
 public:
     virtual ~LineReader() = default;
     virtual Status read_line(const uint8_t** ptr, size_t* size, bool* eof,
-                             const io::IOContext* io_ctx) = 0;
+                             const io::IOContext* io_ctx, size_t* read_bytes = nullptr) = 0;
```
Comment (yiguolei) [inline be/src/exec/line_reader.h:31]: I think we should first refactor broker load and use insert into select to implement broker load. And then we add scan bytes to all scan node . In this way, we will add progress feature to all load types, like stream load, broker load, tvf load, insert into select

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java [hunk 3]
```diff
@@ -364,9 +364,16 @@ private void updateLoadingStatus(BrokerLoadingTaskAttachment attachment) {
 
     @Override
     public void updateProgress(Long beId, TUniqueId loadId, TUniqueId fragmentId, long scannedRows,
-                               long scannedBytes, boolean isDone) {
-        super.updateProgress(beId, loadId, fragmentId, scannedRows, scannedBytes, isDone);
-        progress = (int) ((double) loadStatistic.getLoadBytes() / loadStatistic.totalFileSizeB * 100);
+                               long scannedBytes, long readBytes, boolean isSupportReadBytes, boolean isDone) {
+        super.updateProgress(beId, loadId, fragmentId, scannedRows, scannedBytes,
+                readBytes, isSupportReadBytes, isDone);
+
```
Comment (freemandealer) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java:370]: we'd better show ReadBytes alongside LoadBytes in `show load` results.

Comment (1330571) [reply fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java:369]: > we'd better show ReadBytes alongside LoadBytes in `show load` results.

add those fields in `getShowInfo` of  `loadJob.java` V2, is this right?

**Pull request closed without merging**